File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    index = 0
    current = 0
    while current <= n and index < len(section):
        nextPaint = section[index]
        index += 1
        if current <= nextPaint:
            current = nextPaint + m
            answer += 1
    return answer 

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
import java.util.*;
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int index = 0;
        int current = 0;
        while(current <= n && index < section.length){
            int nextPaint = section[index];
            index++;
            if(current <= nextPaint){
                current = nextPaint;
                current += m;
                answer++;
            }
        }
        return answer;
    }
}